commcare_connect/opportunity/views.py

```diff
@@ -429,18 +429,18 @@ def download_export(request, org_slug, task_id):
 def update_visit_status_import(request, org_slug=None, pk=None):
     opportunity = get_opportunity_or_404(org_slug=org_slug, pk=pk)
     file = request.FILES.get("visits")
-    try:
-        status = bulk_update_visit_status(opportunity, file)
-    except ImportException as e:
-        messages.error(request, e.message)
+    redirect_url_slug = "opportunity:user_visit_review" if opportunity.managed else "opportunity:detail"
+    redirect_url = reverse(redirect_url_slug, args=(request.org.slug, pk))
+
+    file_format = get_file_extension(file)
+    if file_format not in ("csv", "xlsx"):
+        messages.error(request, f"Invalid file format. Only 'CSV' and 'XLSX' are supported. Got {file_format}")
+        return redirect(redirect_url)
     else:
-        message = f"Visit status updated successfully for {len(status)} visits."
-        if status.missing_visits:
-            message += status.get_missing_message()
-        messages.success(request, mark_safe(message))
-    if opportunity.managed:
-        return redirect("opportunity:user_visit_review", org_slug, pk)
-    return redirect("opportunity:detail", org_slug, pk)
+        imported_data = get_imported_dataset(file, file_format)
+        rows = list(imported_data)
+        result = bulk_update_visit_status_task.delay(opportunity.pk, imported_data.headers or [], rows)
```

Comment:
same comment about passing rows here as payment, but blocking in this case. I think given the size visit imports can be, having a hard limit, and ideally not passing the rows at all is important.